aio_pika/queue.py

```diff
@@ -441,32 +441,24 @@ async def close(self, *_: Any) -> Any:
 
         log.debug("Queue iterator %r closed", self)
 
-        # Reject all messages
-        msg: Optional[IncomingMessage] = None
-        try:
-            while True:
-                msg = self._queue.get_nowait()
-        except asyncio.QueueEmpty:
-            if msg is None:
-                return
+        # Reject all messages that have been received and in the buffer/cache.
```

Comment:
While I fixed the code here locally as I thought it makes sense, something does still is uneasy with me. I would expect that the code here is not necessary and may even contain a liveliness issue.

Regarding the liveliness issue: Assume a queue that is receiving messages rapidly. The 'while' loop would be stuck manually nack'ing each and every single message until it is fast enough to escape the 'while' loop. It is possible this never happens. Also, it is possible that messages are still received after escaping the 'while' loop but just before the subscription termination message is send to RabbitMQ. These messages are not nack'ed.

Regarding the necessity: If an application wants each message to arrive, the message would not be automatically ack'ed but rather manually ack'ed after the message is persisted by whatever application integrates aio-pika. If a different delivery guarantee is expected, I would expect the application to allow for messages being dropped or duplicated. In other words, if the local storage is not explicitely ack'ed, the whole system should still work 'as expected'. ('As expected' here means as configured. If is a message is auto-acked (at-most-once guarantee), the application should be able to deal with missing messages. If at-least-once or exactly-once is expected, RabbitMQ should just requeue the moment the connection is dropped even if we do not requeue the messages).

So I am a bit confused as to why this code is required. @mosquito  Perhaps you could elaborate / provide guidance? Thanks in advance!